index.d.ts

```diff
@@ -8,18 +8,22 @@ export interface StoreCreator {
   ): Store<S>;
 }
 
-export type Loop<S, A extends Action> = [S, CmdType<A>];
+export type Loop<S, A extends Action = never> = [S, CmdType<A>];
 
-export interface LoopReducer<S, A extends Action = AnyAction> {
+export interface LoopReducerDeprecated<S, A extends Action = AnyAction> {
   (state: S | undefined, action: A, ...args: any[]): S | Loop<S, A>;
 }
 
-export interface LoopReducerWithDefinedState<S, A extends Action = AnyAction> {
-  (state: S, action: A, ...args: any[]): S | Loop<S, A>;
+export interface LoopReducer<S, LoopActions extends Action = never> {
+  (state: S | undefined, action: AnyAction, ...args: any[]): S | Loop<S, LoopActions>;
 }
 
-export interface LiftedLoopReducer<S, A extends Action = AnyAction> {
-  (state: S | undefined, action: A, ...args: any[]): Loop<S, A>;
+export interface LoopReducerWithDefinedState<S, A extends Action = AnyAction, B extends Action = A> {
+  (state: S, action: A, ...args: any[]): S | Loop<S, B>;
+}
+
+export interface LiftedLoopReducer<S, A extends Action = AnyAction, B extends Action = A> {
```

Comment:
same here

Reply:
oh actually this one should just be updated to the new pattern and we can rename this one deprecated